components/remote_settings/src/cache.rs

```diff
@@ -0,0 +1,259 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use std::{
+    collections::HashSet,
+    fs::{File, OpenOptions},
+    io::prelude::*,
+};
+
+use parking_lot::Mutex;
+
+use crate::{RemoteSettingsResponse, Result};
+
+/// Interface for caching remote settings data
+///
+/// This is implemented by the remote settings consumer, sometimes with foreign code via UniFFI.
+///
+/// This interface has no error handling.  Consumers should report any errors themselves.  If there
+/// is an error in `get()` simply return `None` and the remote settings client will re-fetch any
+/// data.
+pub trait RemoteSettingsCache: Send + Sync {
+    /// Update the cached value
+    fn store(&self, value: RemoteSettingsResponse);
```

Comment:
Wouldn't it be useful to provide `clear()`? Two potential uses I can think of:
- To clear out the cache if the user disables the feature
- For cache busting, sometimes we want the component to ignore the cache and re-download everything